File: dota-zone/dotazone/src/main/java/br/com/dotazone/DotazoneMenu.java
package br.com.dotazone;

import android.content.Intent;
import android.graphics.Color;
import android.graphics.Typeface;
import android.support.v4.app.FragmentActivity;
import android.support.v4.widget.DrawerLayout;
import android.view.View;
import android.view.View.OnClickListener;
import android.widget.Button;
import android.widget.FrameLayout;
import android.widget.RelativeLayout;
import android.widget.TextView;

import br.com.dotazone.model.billing.FeaturePremium;
import br.com.dotazone.model.listeners.Controllable;
import br.com.dotazone.view.activity.AboutActivity;
import br.com.dotazone.view.activity.LanguageActivity;
import br.com.dotazone.view.activity.MainActivity;
import br.com.dotazone.view.activity.PaymentActivity;
import br.com.dotazone.view.activity.TabActivity;

public class DotazoneMenu implements Controllable {

    public static final int MENU_HERO = 0;
    public static final int MENU_ITEM = 1;
    public static final int MENU_LANGUAGE = 2;
    public static final int MENU_BUILD = 3;
    public static final int MENU_NEWS = 4;
    public DrawerLayout drawerLayout;
    public RelativeLayout drawerList;
    private TextView mTextMenuHero;
    private TextView mTextMenuItems;
    private TextView mTextMenuNews;
    private TextView mTextMenuInvite;
    private TextView mTextMenuLanguage;
    private TextView mTextMenuBuild;
    private FragmentActivity mActivity;
    private TextView mtextMenuAbout;
    private Button mPaymentButton;
    private FrameLayout mFrameAdPro;

    public DotazoneMenu(FragmentActivity activity, DrawerLayout drawerLayout, RelativeLayout drawerList) {

        this.mActivity = activity;

        initComponents();

    }

    /**
     * Diplaying fragment view for selected nav drawer list item
     */
    private void displayView(int position) {

        Intent intent = new Intent(mActivity, TabActivity.class);

        switch (position) {
            case 0:
                Intent intentNews = new Intent(mActivity, MainActivity.class);
                intentNews.putExtra(MainActivity.class.getName(), MENU_NEWS);
                mActivity.startActivity(intentNews);
                break;
            case 1:
                intent.putExtra(MainActivity.class.getName(), MENU_HERO);
                mActivity.startActivity(intent);
                TabActivity.isBuild = false;

                break;
            case 2:
                intent.putExtra(MainActivity.class.getName(), MENU_ITEM);
                mActivity.startActivity(intent);
                break;

            case 5:
                mActivity.startActivity(new Intent(mActivity, LanguageActivity.class));
                break;

            case 6:
                intent.putExtra(MainActivity.class.getName(), MENU_BUILD);
                mActivity.startActivity(intent);
                TabActivity.isBuild = true;
                break;
            case 7:

                mActivity.startActivity(new Intent(mActivity, AboutActivity.class));
                break;

            default:
                break;
        }

        drawerLayout.closeDrawer(drawerList);
    }

    public void defaultItemSelected() {
        mTextMenuNews.setTextColor(Color.RED);
    }

    public void checkHeroMenu() {

        mTextMenuItems.setTextColor(Color.WHITE);
        mTextMenuHero.setTextColor(Color.RED);
        mTextMenuNews.setTextColor(Color.WHITE);
        mTextMenuLanguage.setTextColor(Color.WHITE);
        mTextMenuInvite.setTextColor(Color.WHITE);
        mtextMenuAbout.setTextColor(Color.WHITE);
    }

    public void checkItemMenu() {

        mTextMenuItems.setTextColor(Color.RED);
        mTextMenuHero.setTextColor(Color.WHITE);
        mTextMenuNews.setTextColor(Color.WHITE);
        mTextMenuLanguage.setTextColor(Color.WHITE);
        mTextMenuInvite.setTextColor(Color.WHITE);
        mtextMenuAbout.setTextColor(Color.WHITE);
    }

    public void checkLanguageMenu() {

        mTextMenuItems.setTextColor(Color.WHITE);
        mTextMenuHero.setTextColor(Color.WHITE);
        mTextMenuNews.setTextColor(Color.WHITE);
        mTextMenuInvite.setTextColor(Color.WHITE);
        mTextMenuLanguage.setTextColor(Color.RED);
        mtextMenuAbout.setTextColor(Color.WHITE);

    }

    public void checkInviteMenu() {

        mTextMenuItems.setTextColor(Color.WHITE);
        mTextMenuHero.setTextColor(Color.WHITE);
        mTextMenuNews.setTextColor(Color.WHITE);
        mTextMenuLanguage.setTextColor(Color.WHITE);
        mTextMenuInvite.setTextColor(Color.RED);
        mtextMenuAbout.setTextColor(Color.WHITE);

    }

    public void checkNewsMenu() {

        mTextMenuItems.setTextColor(Color.WHITE);
        mTextMenuHero.setTextColor(Color.WHITE);
        mTextMenuLanguage.setTextColor(Color.WHITE);
        mTextMenuNews.setTextColor(Color.RED);
        mTextMenuInvite.setTextColor(Color.WHITE);
        mtextMenuAbout.setTextColor(Color.WHITE);
    }

    public void checkBuildMenu() {

        mTextMenuItems.setTextColor(Color.WHITE);
        mTextMenuHero.setTextColor(Color.WHITE);
        mTextMenuNews.setTextColor(Color.WHITE);
        mTextMenuInvite.setTextColor(Color.WHITE);
        mTextMenuLanguage.setTextColor(Color.WHITE);
        mTextMenuBuild.setTextColor(Color.RED);
        mtextMenuAbout.setTextColor(Color.WHITE);
    }

    public void checkAboutMenu() {

        mTextMenuItems.setTextColor(Color.WHITE);
        mTextMenuHero.setTextColor(Color.WHITE);
        mTextMenuNews.setTextColor(Color.WHITE);
        mTextMenuInvite.setTextColor(Color.WHITE);
        mTextMenuLanguage.setTextColor(Color.WHITE);
        mTextMenuBuild.setTextColor(Color.WHITE);
        mtextMenuAbout.setTextColor(Color.RED);
    }

    public void defaultMenu() {

        mTextMenuItems.setTextColor(Color.WHITE);
        mTextMenuHero.setTextColor(Color.WHITE);
        mTextMenuNews.setTextColor(Color.WHITE);
        mTextMenuInvite.setTextColor(Color.WHITE);
        mTextMenuLanguage.setTextColor(Color.WHITE);
        mTextMenuBuild.setTextColor(Color.WHITE);
        mtextMenuAbout.setTextColor(Color.WHITE);
    }

    public void initComponents() {

        Typeface font = Typeface.createFromAsset(mActivity.getAssets(), "Roboto-Thin.ttf");

        drawerLayout = (DrawerLayout) mActivity.findViewById(R.id.drawer_layout);
        drawerList = (RelativeLayout) mActivity.findViewById(R.id.list_slidermenu);
        mTextMenuHero = (TextView) mActivity.findViewById(R.id.textMenuHero);
        mFrameAdPro = (FrameLayout) mActivity.findViewById(R.id.content_drawer_list_account_premium);
        mTextMenuHero.setOnClickListener(new OnClickListener() {

            @Override
            public void onClick(View v) {

                checkHeroMenu();
                displayView(1);

            }
        });
        mTextMenuItems = (TextView) mActivity.findViewById(R.id.textMenuItems);
        mTextMenuItems.setOnClickListener(new OnClickListener() {

            @Override
            public void onClick(View v) {

                checkItemMenu();
                displayView(2);
            }
        });
        mTextMenuNews = (TextView) mActivity.findViewById(R.id.textMenuNews);
        mTextMenuNews.setOnClickListener(new OnClickListener() {

            @Override
            public void onClick(View v) {

                checkNewsMenu();
                displayView(0);

            }
        });
        mTextMenuInvite = (TextView) mActivity.findViewById(R.id.textMenuInvite);
        mTextMenuInvite.setOnClickListener(new OnClickListener() {

            @Override
            public void onClick(View v) {

                checkInviteMenu();
                Intent sharingIntent = new Intent(android.content.Intent.ACTION_SEND);
                sharingIntent.setType("text/plain");
                String shareBody = mActivity.getString(R.string.share_app_text);
                sharingIntent.putExtra(android.content.Intent.EXTRA_TEXT, shareBody);
                mActivity.startActivity(Intent.createChooser(sharingIntent, "Share via"));

            }
        });

        mTextMenuLanguage = (TextView) mActivity.findViewById(R.id.textMenuLanguage);
        mTextMenuLanguage.setOnClickListener(new OnClickListener() {

            @Override
            public void onClick(View v) {

                checkLanguageMenu();
                displayView(5);

            }
        });

        mTextMenuBuild = (TextView) mActivity.findViewById(R.id.textMenuBuild);
        mTextMenuBuild.setOnClickListener(new OnClickListener() {

            @Override
            public void onClick(View v) {

                checkBuildMenu();
                displayView(6);

            }
        });

        mtextMenuAbout = (TextView) mActivity.findViewById(R.id.textMenuAbout);
        mtextMenuAbout.setOnClickListener(new OnClickListener() {

            @Override
            public void onClick(View v) {
                checkAboutMenu();
                displayView(7);

            }
        });

        mtextMenuAbout.setTypeface(font);
        mTextMenuHero.setTypeface(font);
        mTextMenuItems.setTypeface(font);
        mTextMenuNews.setTypeface(font);
        mTextMenuInvite.setTypeface(font);
        mTextMenuLanguage.setTypeface(font);
        mTextMenuBuild.setTypeface(font);

        mPaymentButton = (Button) mActivity.findViewById(R.id.content_drawer_buy_button);
        mPaymentButton.setOnClickListener(new OnClickListener() {

            @Override
            public void onClick(View v) {


                mActivity.startActivity(new Intent(mActivity, PaymentActivity.class));

            }
        });

        defaultItemSelected();

        FeaturePremium featurePremium = new FeaturePremium(mActivity);
        featurePremium.unlockOrLockAdPro(mPaymentButton, mFrameAdPro, DotaZoneBrain.isPremium);
    }

    @Override
    public void menu(boolean isOpen) {

        drawerLayout.openDrawer(drawerList);

    }


}

File: dota-zone/app/src/main/java/br/com/dotazone/DotaZoneMenu.kt
package br.com.dotazone

import android.content.Intent
import android.graphics.Color
import android.graphics.Typeface
import android.view.View
import android.widget.Button
import android.widget.FrameLayout
import android.widget.RelativeLayout
import android.widget.TextView
import androidx.drawerlayout.widget.DrawerLayout
import androidx.fragment.app.FragmentActivity
import br.com.dotazone.model.listeners.Controllable
import br.com.dotazone.view.activity.AboutActivity
import br.com.dotazone.view.activity.LanguageActivity
import br.com.dotazone.view.activity.MainActivity
import br.com.dotazone.view.activity.TabActivity


class DotazoneMenu(private val mActivity: FragmentActivity, drawerLayout: DrawerLayout?, drawerList: RelativeLayout?) : Controllable {
	var drawerLayout: DrawerLayout? = null
	var drawerList: RelativeLayout? = null
	private var mTextMenuHero: TextView? = null
	private var mTextMenuItems: TextView? = null
	private var mTextMenuNews: TextView? = null
	private var mTextMenuInvite: TextView? = null
	private var mTextMenuLanguage: TextView? = null
	private var mTextMenuBuild: TextView? = null
	private var mTextMenuChannel: TextView? = null
	private var mtextMenuAbout: TextView? = null
	private var mPaymentButton: Button? = null
	private var mFrameAdPro: FrameLayout? = null

	/**
	 * Diplaying fragment view for selected nav drawer list item
	 */
	private fun displayView(position: Int) {
		val intent = Intent(mActivity, TabActivity::class.java)
		when (position) {
			0 -> {
				val intentNews = Intent(mActivity, MainActivity::class.java)
				intentNews.putExtra(MainActivity::class.java.name, MENU_NEWS)
				mActivity.startActivity(intentNews)
			}
			1 -> {
				intent.putExtra(MainActivity::class.java.name, MENU_HERO)
				mActivity.startActivity(intent)
				TabActivity.isBuild = false
			}
			2 -> {
				intent.putExtra(MainActivity::class.java.name, MENU_ITEM)
				mActivity.startActivity(intent)
			}
			5 -> mActivity.startActivity(Intent(mActivity, LanguageActivity::class.java))
			6 -> {
				intent.putExtra(MainActivity::class.java.name, MENU_BUILD)
				mActivity.startActivity(intent)
				TabActivity.isBuild = true
			}
			7 -> mActivity.startActivity(Intent(mActivity, AboutActivity::class.java))
			else -> {
			}
		}
		drawerLayout!!.closeDrawer(drawerList!!)
	}

	fun defaultItemSelected() {
		mTextMenuNews!!.setTextColor(Color.RED)
	}

	fun checkHeroMenu() {
		mTextMenuItems!!.setTextColor(Color.WHITE)
		mTextMenuHero!!.setTextColor(Color.RED)
		mTextMenuNews!!.setTextColor(Color.WHITE)
		mTextMenuLanguage!!.setTextColor(Color.WHITE)
		mTextMenuInvite!!.setTextColor(Color.WHITE)
		mtextMenuAbout!!.setTextColor(Color.WHITE)
	}

	fun checkItemMenu() {
		mTextMenuItems!!.setTextColor(Color.RED)
		mTextMenuHero!!.setTextColor(Color.WHITE)
		mTextMenuNews!!.setTextColor(Color.WHITE)
		mTextMenuLanguage!!.setTextColor(Color.WHITE)
		mTextMenuInvite!!.setTextColor(Color.WHITE)
		mtextMenuAbout!!.setTextColor(Color.WHITE)
	}

	fun checkLanguageMenu() {
		mTextMenuItems!!.setTextColor(Color.WHITE)
		mTextMenuHero!!.setTextColor(Color.WHITE)
		mTextMenuNews!!.setTextColor(Color.WHITE)
		mTextMenuInvite!!.setTextColor(Color.WHITE)
		mTextMenuLanguage!!.setTextColor(Color.RED)
		mtextMenuAbout!!.setTextColor(Color.WHITE)
	}

	fun checkInviteMenu() {
		mTextMenuItems!!.setTextColor(Color.WHITE)
		mTextMenuHero!!.setTextColor(Color.WHITE)
		mTextMenuNews!!.setTextColor(Color.WHITE)
		mTextMenuLanguage!!.setTextColor(Color.WHITE)
		mTextMenuInvite!!.setTextColor(Color.RED)
		mtextMenuAbout!!.setTextColor(Color.WHITE)
	}

	fun checkNewsMenu() {
		mTextMenuItems!!.setTextColor(Color.WHITE)
		mTextMenuHero!!.setTextColor(Color.WHITE)
		mTextMenuLanguage!!.setTextColor(Color.WHITE)
		mTextMenuNews!!.setTextColor(Color.RED)
		mTextMenuInvite!!.setTextColor(Color.WHITE)
		mtextMenuAbout!!.setTextColor(Color.WHITE)
	}

	fun checkBuildMenu() {
		mTextMenuItems!!.setTextColor(Color.WHITE)
		mTextMenuHero!!.setTextColor(Color.WHITE)
		mTextMenuNews!!.setTextColor(Color.WHITE)
		mTextMenuInvite!!.setTextColor(Color.WHITE)
		mTextMenuLanguage!!.setTextColor(Color.WHITE)
		mTextMenuBuild!!.setTextColor(Color.RED)
		mtextMenuAbout!!.setTextColor(Color.WHITE)
	}

	fun checkAboutMenu() {
		mTextMenuItems!!.setTextColor(Color.WHITE)
		mTextMenuHero!!.setTextColor(Color.WHITE)
		mTextMenuNews!!.setTextColor(Color.WHITE)
		mTextMenuInvite!!.setTextColor(Color.WHITE)
		mTextMenuLanguage!!.setTextColor(Color.WHITE)
		mTextMenuBuild!!.setTextColor(Color.WHITE)
		mtextMenuAbout!!.setTextColor(Color.RED)
	}

	fun checkChannelMenu() {
		mTextMenuItems!!.setTextColor(Color.WHITE)
		mTextMenuHero!!.setTextColor(Color.WHITE)
		mTextMenuNews!!.setTextColor(Color.WHITE)
		mTextMenuInvite!!.setTextColor(Color.WHITE)
		mTextMenuLanguage!!.setTextColor(Color.WHITE)
		mTextMenuBuild!!.setTextColor(Color.WHITE)
		mtextMenuAbout!!.setTextColor(Color.WHITE)
		mTextMenuChannel!!.setTextColor(Color.RED)
	}

	fun defaultMenu() {
		mTextMenuItems!!.setTextColor(Color.WHITE)
		mTextMenuHero!!.setTextColor(Color.WHITE)
		mTextMenuNews!!.setTextColor(Color.WHITE)
		mTextMenuInvite!!.setTextColor(Color.WHITE)
		mTextMenuLanguage!!.setTextColor(Color.WHITE)
		mTextMenuBuild!!.setTextColor(Color.WHITE)
		mtextMenuAbout!!.setTextColor(Color.WHITE)
		mTextMenuChannel!!.setTextColor(Color.WHITE)
	}

	fun initComponents() {
		val font = Typeface.createFromAsset(mActivity.assets, "Roboto-Thin.ttf")
		drawerLayout = mActivity.findViewById<View>(R.id.drawer_layout) as DrawerLayout
		drawerList = mActivity.findViewById<View>(R.id.listSliderMenu) as RelativeLayout
		mTextMenuHero = mActivity.findViewById<View>(R.id.textMenuHero) as TextView
		mFrameAdPro = mActivity.findViewById<View>(R.id.content_drawer_list_account_premium) as FrameLayout
		mTextMenuHero!!.setOnClickListener {
			checkHeroMenu()
			displayView(1)
		}
		mTextMenuItems = mActivity.findViewById<View>(R.id.textMenuItems) as TextView
		mTextMenuItems!!.setOnClickListener {
			checkItemMenu()
			displayView(2)
		}
		mTextMenuNews = mActivity.findViewById<View>(R.id.textMenuNews) as TextView
		mTextMenuNews!!.setOnClickListener {
			checkNewsMenu()
			displayView(0)
		}
		mTextMenuInvite = mActivity.findViewById<View>(R.id.textMenuInvite) as TextView
		mTextMenuInvite!!.setOnClickListener {
			checkInviteMenu()
			val sharingIntent = Intent(Intent.ACTION_SEND)
			sharingIntent.type = "text/plain"
			val shareBody = mActivity.getString(R.string.share_app_text)
			sharingIntent.putExtra(Intent.EXTRA_TEXT, shareBody)
			mActivity.startActivity(Intent.createChooser(sharingIntent, "Share via"))
		}
		mTextMenuLanguage = mActivity.findViewById<View>(R.id.textMenuLanguage) as TextView
		mTextMenuLanguage!!.setOnClickListener {
			checkLanguageMenu()
			displayView(5)
		}
		mTextMenuBuild = mActivity.findViewById<View>(R.id.textMenuBuild) as TextView
		mTextMenuBuild!!.setOnClickListener {
			checkBuildMenu()
			displayView(6)
		}
		mtextMenuAbout = mActivity.findViewById<View>(R.id.textMenuAbout) as TextView
		mtextMenuAbout!!.setOnClickListener {
			checkAboutMenu()
			displayView(7)
		}
		mTextMenuChannel = mActivity.findViewById<View>(R.id.textMenuChannel) as TextView
		mTextMenuChannel!!.setOnClickListener {
			checkChannelMenu()
			displayView(8)
		}
		mtextMenuAbout!!.typeface = font
		mTextMenuHero!!.typeface = font
		mTextMenuItems!!.typeface = font
		mTextMenuNews!!.typeface = font
		mTextMenuInvite!!.typeface = font
		mTextMenuLanguage!!.typeface = font
		mTextMenuBuild!!.typeface = font
		mTextMenuChannel!!.typeface = font
		mPaymentButton = mActivity.findViewById<View>(R.id.content_drawer_buy_button) as Button
		mPaymentButton!!.setOnClickListener {
			//mActivity.startActivity(new Intent(mActivity, PaymentActivity.class));
		}
		defaultItemSelected()
	}

	override fun menu(isOpen: Boolean) {
		drawerLayout!!.openDrawer(drawerList!!)
	}

	companion object {
		const val MENU_HERO = 0
		const val MENU_ITEM = 1
		const val MENU_LANGUAGE = 2
		const val MENU_BUILD = 3
		const val MENU_NEWS = 4
	}

	init {
		initComponents()
	}
}

